document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresQueryParser.java

```diff
@@ -1,149 +1,43 @@
 package org.hypertrace.core.documentstore.postgres;
 
-import static org.hypertrace.core.documentstore.Collection.UNSUPPORTED_QUERY_OPERATION;
-import static org.hypertrace.core.documentstore.postgres.PostgresCollection.CREATED_AT;
-import static org.hypertrace.core.documentstore.postgres.PostgresCollection.DOCUMENT;
-import static org.hypertrace.core.documentstore.postgres.PostgresCollection.DOC_PATH_SEPARATOR;
-import static org.hypertrace.core.documentstore.postgres.PostgresCollection.ID;
-import static org.hypertrace.core.documentstore.postgres.PostgresCollection.UPDATED_AT;
-
 import java.util.Arrays;
-import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
+import java.util.Optional;
 import java.util.stream.Collectors;
 import org.apache.commons.lang3.StringUtils;
 import org.hypertrace.core.documentstore.Filter;
 import org.hypertrace.core.documentstore.OrderBy;
 import org.hypertrace.core.documentstore.postgres.Params.Builder;
+import org.hypertrace.core.documentstore.postgres.utils.PostgresUtils;
 
 class PostgresQueryParser {
 
-  private static final String QUESTION_MARK = "?";
-  // postgres jsonb uses `->` instead of `.` for json field access
-  private static final String JSON_FIELD_ACCESSOR = "->";
-  // postgres operator to fetch the value of json object as text.
-  private static final String JSON_DATA_ACCESSOR = "->>";
-  private static final Set<String> OUTER_COLUMNS =
-      new HashSet<>() {
-        {
-          add(CREATED_AT);
-          add(ID);
-          add(UPDATED_AT);
-        }
-      };
+  static String parseSelections(List<String> selections) {
+    return Optional.of(
+            selections.stream()
+                .map(
+                    selection ->
+                        String.format(
+                            "%s AS \"%s\"",
+                            PostgresUtils.prepareFieldAccessorExpr(
+                                selection, PostgresUtils.DOCUMENT_COLUMN),
+                            selection))
+                .collect(Collectors.joining(",")))
+        .filter(str -> StringUtils.isNotBlank(str))
```

Comment:
Nit: Can be simplified as 
```suggestion
        .filter(StringUtils::isNotBlank)
```